### PR TITLE
#153 - Add code coverage support improve

### LINKED ISSUE #153: Improve code-coverage & add GH check action
## Background
Improve or add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.
Add GH action to check changed file coverage.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Used Absa fork of JaCoCo solution - with scala method filtering.
- Added support for code coverage GitHub action check as part of PR.

Closes #153

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -19,6 +19,23 @@ addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.6.0")
...
-addSbtPlugin("com.github.sbt" % "sbt-jacoco" % "3.4.0")
+// sbt-jacoco - workaround related dependencies required to download
+lazy val ow2Version = "9.5"
+lazy val jacocoVersion = "0.8.9-absa.1"
+
+def jacocoUrl(artifactName: String): String = s"https://github.com/AbsaOSS/jacoco/releases/download/$jacocoVersion/org.jacoco.$artifactName-$jacocoVersion.jar"
+def ow2Url(artifactName: String): String = s"https://repo1.maven.org/maven2/org/ow2/asm/$artifactName/$ow2Version/$artifactName-$ow2Version.jar"
+
+addSbtPlugin("com.jsuereth" %% "scala-arm" % "2.0" from "https://repo1.maven.org/maven2/com/jsuereth/scala-arm_2.11/2.0/scala-arm_2.11-2.0.jar")
+addSbtPlugin("com.jsuereth" %% "scala-arm" % "2.0" from "https://repo1.maven.org/maven2/com/jsuereth/scala-arm_2.12/2.0/scala-arm_2.12-2.0.jar")
```
Comment (dk1844) [inline project/plugins.sbt:30]: Isn't this a problem - adding both plugins for Scala 2.11 and for 2.12 regardless of which is used?

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code)

**Review by Zejnilovic — APPROVED** [review]
Failing of Spark 3 is "normal" in a sense that it is a known issue.

File: project/plugins.sbt (hunk 1 above)
Comment (miroslavpojer) [reply project/plugins.sbt:30]: It looks but it works.

**Review by dk1844 — APPROVED** [review]

**Pull request merged**